autolab/Autolab
# Add first batch of submissions tests, refactor backend code to have more error handling / validation

## Description


1. Remove the `#show` route, which is unused, from `routes.rb` and `submissions_controller.rb`
2. Refactor `#new` to have better and correct error-checking so that a 500 error wouldn't occur when a CUD ID didn't exist in the list of CUD IDs provided to the action
3. Add validation of params / error-checking in `#create` on invalid submission creation
4. Add similar error-checking to `#update`
6. Update the submission destroy interface to be more user-friendly and make more sense, by disabling the button to destroy the submission unless the instructor has checked the box indicating the read the disclaimer and still wish to destroy the submission. Made buttons and checkbox more descriptive as well.

Before: Instructor could still click button to destroy the submission without checking the box but would get an "error" back.

<img width="1189" alt="Screenshot 2024-03-08 at 6 22 03 PM" src="https://github.com/autolab/Autolab/assets/25730111/07f825c7-8d7e-45d2-8cac-813c09e6844e">
<img width="1106" alt="Screenshot 2024-03-08 at 6 22 08 PM" src="https://github.com/autolab/Autolab/assets/25730111/1310e6ec-3c6f-45e3-b0b1-2994471eb63b">

After: Instructor can't click button without having checked the box first.

<img width="1067" alt="Screenshot 2024-03-08 at 6 20 49 PM" src="https://github.com/autolab/Autolab/assets/25730111/c74e177f-8708-48f5-9367-1aa3ec1a3fab">
<img width="1055" alt="Screenshot 2024-03-08 at 6 21 00 PM" src="https://github.com/autolab/Autolab/assets/25730111/4ba9b7a2-9662-469a-ae5a-336df167a495">

7. Add tests for:
- new
- create
- edit

(also try to fix flaky assessment_spec test by giving capybara expect more wait time)

8. Fix version number not decreasing when assessment is deleted
- add method `delete_version_number` to decrease version number when submission is destroyed, and update `update_latest_submission` so that it finds the latest submission based on creation date (and checks with version number)

## Motivation and Context



Fixes #2109 

## How Has This Been Tested?




- Try to create submission for CUD with ids via url e.g. `http://localhost:3000/courses/AutoPopulated/assessments/lab5/submissions/new?course_user_datum_id=29,30`, see that it renders correctly
- try to do some with non-existent CUD id, see that error appears

<img width="833" alt="Screenshot 2024-03-08 at 7 12 49 PM" src="https://github.com/autolab/Autolab/assets/25730111/219888e4-ec24-4f00-8884-df0d2fd19792">

- verify that creating missing submissions still work
- Run through destroy submission flow, see that it works and makes sense
- Create submissions for a user (can be yourself), then destroy them, then create more submissions, see that submission number updates correctly
- Run tests locally and see that they pass


## Types of changes



## Checklist:
